coderd/notifications/manager.go

```diff
@@ -0,0 +1,450 @@
+package notifications
+
+import (
+	"context"
+	"encoding/json"
+	"sync"
+	"time"
+
+	"github.com/google/uuid"
+	"golang.org/x/sync/errgroup"
+	"golang.org/x/xerrors"
+
+	"github.com/coder/coder/v2/coderd/notifications/render"
+	"github.com/coder/coder/v2/codersdk"
+
+	"github.com/coder/coder/v2/coderd/database"
+	"github.com/coder/coder/v2/coderd/notifications/dispatch"
+	"github.com/coder/coder/v2/coderd/notifications/types"
+
+	"cdr.dev/slog"
+)
+
+// Manager manages all notifications being enqueued and dispatched.
+//
+// Manager maintains a group of notifiers: these consume the queue of notification messages in the store.
+//
+// Notifiers dequeue messages from the store _CODER_NOTIFICATIONS_LEASE_COUNT_ at a time and concurrently "dispatch" these messages, meaning they are
+// sent by their respective methods (email, webhook, etc).
```

Comment:
I don't follow.  If the plan is separate them for scale, then they don't need to handle memory concurrency.

Reply:
We spoke offline and I was missing your point: which was that we needn't start multiple notifiers since we dispatch messages concurrently. I've implemented this change in 9380d8e6ee0e89b68a5a71dd352c37a447c48f04.